Alipay Robot Code/lib/Melty/melty.cpp

```diff
@@ -0,0 +1,90 @@
+#include <melty.h>
+#include <LEDHandler.h>
+
+melty::melty() {
+    period_micros_calc = ringBuffer(1);
+    time_seen_beacon_calc = ringBuffer(0.5); 
+}
+
+void melty::update() {
+    bool curSeenIRLed;
+    if (useTopIr)
```

Comment:
Recommend always using curly braces for readability https://google.github.io/styleguide/cppguide.html#Formatting_Looping_Branching